### ISSUE TITLE
Activating Google Tag Manager container for wellcomecollection.org

### ISSUE THREAD

**Issue by taceybadgerbrook** [wellcomecollection/wellcomecollection.org#8524]
Google Tag Manager was used previously for tagging data across wellcomecollection.org and wellcomelibrary.org. The container snippets should be deleted and replaced with a new snippet.

The old snippet in the head was:

```
<!-- Google Tag Manager -->
<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','GTM-WV9JS8W');</script>
<!-- End Google Tag Manager -->
```

The old snippet in the body was:

```
<!-- Google Tag Manager (noscript) -->
<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-WV9JS8W"
height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
<!-- End Google Tag Manager (noscript) -->
```

Here is a link to the snippets to install for the new wc.org only container: https://tagmanager.google.com/?authuser=0#/admin/accounts/6000872529/containers/31009043/install

**Comment by taceybadgerbrook** [comment]
In order to use Google Tag Manager, a data layer will need to be set up. Here is a link to the snippet for this:
https://developers.google.com/tag-platform/tag-manager/web/datalayer?hl=en_US

**Comment by jamieparkinson** [comment]
Moving this back to the backlog for now as it's a large (and important) change, see thread in Slack https://wellcome.slack.com/archives/C3TQSF63C/p1664797960502819

**Comment by gestchild** [comment]
### Thinking out loud

We currently use Universal Analytics and GA4 without Google Tag Manager and could continue to do so. However, we've agreed that Google Tag Manager will provide certain benefits that make it worth adopting. Not least that it empowers Tacey to set up and remove event tracking without the need for developers to be involved, and all that that entails.

The recommended way of using Google Tag Manager, with Google Analytics, is to install Google Tag Manager and then use it to install Google Analytics. However, we already include the google analytics script (gtag.js) for both our GA4 and Universal Analytics, and track certain events, within the website code.

In order that we don't duplicate and/or lose data, we need to proceed carefully.

I can see a few approaches (option 3 being my favourite).

### 1. Only use Google Tag Manager
We add the Google Tag Manager code to the website and remove the Google Analytics code at the same time.
At present the Google Tag Manager account seems to be set up to include the Universal Analytics and GA4 tags:

![Screenshot 2023-01-23 at 17 09 57](https://user-images.githubusercontent.com/6051896/214281113-be104e8f-0f17-4437-ae1c-b690e0e96ccb.png)

This would be the most risky option, in terms of losing data. For example, if the UA and GA tags in Google Tag Manager don't work as expected. We also have a lot of [code within the application for tracking events](https://docs.google.com/document/d/1m0ARLrfGtNA4dSHq10hpIRvOnwlExnuK62T4ADxPNKo/edit#heading=h.kxtwi6ijy6xa) and I don't know how these would be affected by doing this. We could remove this event tracking code too and try and do it in Tag Manager instead, but that seems like a lot of moving parts and may not be possible for certain things like the performance metrics.

### 2. Take a short term hybrid approach
It is possible to include Google Tag Manager along side Google Analytics in the website code. In this case we would need to be sure that we didn't double track things.

For example, we'd need to remove the UA and GA4 tags from our Google Tag Manager account before we implement Google Tag Manager on the website. We'd also need to make sure we didn't imnplement event tracking via Google Tag Manager for things that are already being tracked in the website code.

We could then move in the direction of removing the analytics code from the website to the scenario of only having the Tag Manager code. We could move the events we care about across to Tag Manager in a more piecemeal fashion. At the same time we could send these events to GA4 rather than UA and remove the events we no longer care about altogether. As a last step we would remove the Google Analytics scripts and implement them in Google Tag Manager and would only need to implement GA4.

### 3. Take a long term hybrid approach
Basically the same as option 2, but with no intention of ever removing the gtag.js analytics scripts from the website. We might also want to keep certain events, like the performance metrics in code. At the moment this is my preferred option.

**Comment by gestchild** [comment]
@jamieparkinson and @taceybadgerbrook I would appreciate your thoughts on the above.

**Comment by taceybadgerbrook** [comment]
@gestchild there are 2 GTM accounts that I can see. [GTM-53DFWQD](https://tagmanager.google.com/) is not a live container. It contains changes but these are the result of me trying to do training on a container that isn't live. None of these changes have been published, so we can throw this away without doing any damage.

The other account is called Wellcome Collection but it has no GTM container.

**Comment by gestchild** [comment]
ok, GTM-53DFWQD was the one linked to at the top of this issue as the one to install. Do we still want to do that? If yes, we'll just need to tidy it up a bit and remove the analytics tags. If no, you'll need to let me know the one you want to use.

Do you know why we have 2 and can we delete one of them?

**Comment by taceybadgerbrook** [comment]
@gestchild Ok, I've cleared out everything in GTM-53DFWQD.

The other account was used to track across wellcomecollection.org and wellcomelibrary.org. The stats it produced looked wrong and so the decision was taken to discontinue use of this. There is no container to delete so I'm not sure how to delete the account.

**Comment by jamieparkinson** [comment]
Oops, yes, will correct original comment

**Comment by gestchild** [comment]
I think they'll be events we'll always need to trigger in code, such as the Next.js [reportWebVitals stuff](https://nextjs.org/docs/advanced-features/measuring-performance#sending-results-to-analytics)

**Comment by jamieparkinson** [comment]
hmmm - on the contrary I think we should be updating and unifying these events, and that this is a good opportunity to do so. eg for web vitals there is a [recommended](https://github.com/GoogleChrome/web-vitals#send-the-results-to-google-tag-manager) way to do it with GA4 and GTM.

Another way to say this - if it's true that we'll always need some events that are triggered in code, I don't think this automatically means we'll need `gtag.js` to do this. Disadvantages of keeping both would obviously be increased cost of maintenance, higher risk of regressing to use gtag for "too much" in future once this context is lost, and increased overhead of JS sent to clients.

**Comment by gestchild** [comment]
That's a fair point and I'm all for moving everything out of the code if we can, I was thinking we'd miss the [Nextjs custom metrics](https://nextjs.org/docs/advanced-features/measuring-performance#custom-metrics), but we don't actually do anything with those currently so maybe that doesn't matter.

**Comment by gestchild** [comment]
@taceybadgerbrook will you be happy setting up [the Web Vitals metric tracking](https://github.com/GoogleChrome/web-vitals#send-the-results-to-google-tag-manager) when the time comes.

**Comment by gestchild** [comment]
If we got with 2, the result of the work means that as well as empowering Tacey to do tracking stuff without developers, it also moves all the responsibility onto those who know how to/want to use Google Tag Manager (which at the moment is @taceybadgerbrook and possibly @jennpb?). I just want to be clear that everyone is happy with that.

**Comment by jennpb** [comment]
**Can I summarise what I think we're agreeing?**

We're moving all tracking to GA4 via GTM including
-  web vitals via the Core Web Vitals custom template mentioned [here](https://github.com/GoogleChrome/web-vitals#send-the-results-to-google-tag-manager)
- component-level events that we want to keep (subject to review with @jennpb @taceybadgerbrook and @DominiqueMarshall )

And that moving the component-level events into GA4 via GTM _can be done independently_ of setting up the GTM container + GA4? 

I have some questions about what out-of-the-box events are already included in GA4, and whether there's duplication of what we currently track?

**Comment by gestchild** [comment]
That all sounds right.

> And that moving the component-level events into GA4 via GTM _can be done independently_ of setting up the GTM container + GA4?

Yes,  I think we'll add GTM, shift over the events, then remove the gtag (analytics code) and implement that in GTM too.

**Comment by gestchild** [comment]
Tag Manager was added in #9174 and we now use GTM to add the GA4 tag
